File: app/src/main/java/com/soutvoid/gamesproject/interactor/common/FilterRequestFormatList.java
package com.soutvoid.gamesproject.interactor.common;

import com.annimon.stream.Collectors;
import com.annimon.stream.Stream;

import java.util.Collection;
import java.util.HashSet;

/**
 * Коллеция из уникальных целочисленных элементов, преобразуемая в строку формата "1,2,3"
 */
public class FilterRequestFormatList extends HashSet<Integer> {

    public FilterRequestFormatList() {
    }

    public FilterRequestFormatList(Collection<Integer> c) {
        super(c);
    }

    @Override
    public String toString() {
        return Stream.of(this)
                .map(Object::toString)
                .collect(Collectors.joining(","));
    }
}

File: app/src/main/java/com/soutvoid/gamesproject/interactor/common/FilterRequestFormatList.kt
package com.soutvoid.gamesproject.interactor.common

import java.util.*

/**
 * Коллеция из уникальных целочисленных элементов, преобразуемая в строку формата "1,2,3"
 */
class FilterRequestFormatList : HashSet<Int> {

    constructor() {}

    constructor(c: Collection<Int>) : super(c) {}

    override fun toString(): String {
        return this.mapIndexed { index, i -> i.toString() }.joinToString(",")
    }
}